address/tests/test_models.py

```diff
@@ -160,6 +160,9 @@ def test_unicode(self):
         self.assertEqual(unicode(self.ad1), u"1 Some Street, Melbourne, Victoria 3000, Australia")
         self.assertEqual(unicode(self.ad_empty), u"Northcote, Victoria 3070, Australia")
 
+    def test_auto_formatted(self):
+        self.assertEqual(str(self.ad1), self.ad1.formatted, self.ad1.formatted)
```

Comment:
As a way to confirm the above comment about `pre_save`/`post_save`, you could try loading the model from the DB in this test before making the change from above. Something like:
```python
self.assertEqual(str(Address.objects.get(id=self.ad1.id)), self.ad1.formatted)
```